<!-- This is an example launch file for the ar_object_tracker calibration process -->
<launch>
  <arg name="debug" default="false"/>
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -rv -e gdb -ex run -args"/>
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>

  <include file="$(find openni_launch)/launch/openni.launch" />

  <node name="AR_Kinect_multi_single" pkg="ar_pose" type="ar_multiple_single"
        respawn="false" output="screen" launch-prefix="$(arg launch_prefix)">
    <remap from="AR_Kinect_multi_single/ar_pose_marker" to="/ar_pose_marker_all" />
    <rosparam command="load" file="$(find ar_object_tracker)/config/ar_multi_single_marker.yaml"/>
    <param name="marker_pattern_list" type="string" value="$(find ar_object_tracker)/config/example_list.dat" />
    <param name="marker_pattern" type="string" value="$(find ar_object_tracker)/config/example_list.dat" />
    <param name="camera_frame" type="string" value="/camera_rgb_optical_frame"/>
    <param name="marker_frame" type="string" value="ARMarker"/>
    <param name="camera_image_topic" type="string" value="/camera/rgb/image_rect_color"/>
    <param name="camera_info_topic" type="string" value="/camera/rgb/camera_info"/>
    <param name="publish_tf" type="bool" value="true"/>
    <param name="marker_red" type="double" value="0.5"/>
    <param name="marker_green" type="double" value="0.0"/>
    <param name="marker_blue" type="double" value="1.0"/>
  </node>

  <arg name="output_file" default="~/.ros/ar_object_tracker/config/example_object.yaml" />

  <node name="ar_calibrate" pkg="ar_object_tracker" type="calibrate" output="screen">
    <param name="output_file" type="string" value="$(arg output_file)" />
    <remap from="ar_calibrate/ar_pose_marker" to="/ar_pose_marker_all" />
  </node>
</launch>
